tst/airfs/airfs.cpp

```diff
@@ -1609,6 +1607,32 @@ NTSTATUS ApiGetStreamInfo(FSP_FILE_SYSTEM *FileSystem, PVOID Node0,
     return STATUS_SUCCESS;
 }
 
+#endif
+//////////////////////////////////////////////////////////////////////
+#if defined(AIRFS_CONTROL)
+
+NTSTATUS ApiControl(FSP_FILE_SYSTEM *FileSystem,
+    PVOID FileContext, UINT32 ControlCode,
+    PVOID InputBuffer, ULONG InputBufferLength,
+    PVOID OutputBuffer, ULONG OutputBufferLength, PULONG PBytesTransferred)
+{
+    //  Trivial example: change upper to lower case and vice versa.
+    if (CTL_CODE(0x8000 + 'A', 'C', METHOD_BUFFERED, FILE_ANY_ACCESS) == ControlCode)
+    {
+        if (OutputBufferLength != InputBufferLength) return STATUS_INVALID_PARAMETER;
+        for (ULONG i = 0; i < InputBufferLength; i++)
+        {
+            char c = ((char*)InputBuffer)[i];
+            if ((c|0x20) >= 'a' && (c|0x20) <= 'z') c ^= 0x20;
+            ((char*)OutputBuffer)[i] = c;
+        }
+        *PBytesTransferred = InputBufferLength;
+        return STATUS_SUCCESS;
+    }
+
+    return STATUS_INVALID_DEVICE_REQUEST;
+}
+
```

Comment:
Thanks for this.

I am wondering though if we should use rot13 to keep it inline with MEMFS. Winfsp-tests currently checks for rot13 in its `DeviceIoControl` test:

https://github.com/billziss-gh/winfsp/blob/master/tst/winfsp-tests/devctl-test.c#L52

Reply:
Of course! And those indent changes are (supposed to be at least) replacing a few tabs that snuck into my code with spaces.

Reply:
Thanks!